### PR TITLE
My first branch

### PR DESCRIPTION
**Pull request by LiviuAC**

### CODE REVIEW

File: test/specs/login.js [hunk 1]
```diff
@@ -53,9 +52,9 @@ xdescribe('Login Test', () => {
         await LoginPage.back()
         const usernameField = await LoginPage.inputUsername.getText()
         const passwordField = await LoginPage.inputPassword.getText()
-        // sa vedem daca logica aceasta este buna sau daca putem folosi isEmpty(). Am incercat si nu mi-a mers
-        expect(usernameField.length).toBeGreaterThan(0)
-        expect(passwordField.length).toBeGreaterThan(0)
+        console.log(`The username field: `, usernameField)
```
Comment (axboory) [inline test/specs/login.js]: remove console.log if not needed anymore.

File: test/specs/login.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 const LoginPage = require('../pageobjects/LoginPage');
 const {CREDENTIALS} = require("../helper/credentials");
 
-xdescribe('Browser Tests', () => {
+describe('Browser Tests', () => {
```
Comment (axboory) [inline test/specs/login.js]: This describe should be renamed 'Login page validation' or 'Login page tests'

File: test/specs/login.js [hunk 3]
```diff
@@ -22,29 +22,28 @@ xdescribe('Browser Tests', () => {
...
-        passwordList.push(passwordText)
-        expect(String(passwordList).replace(/^\[(.+)\]$/,'$1')).toEqual("Password for" +
-            " all users:\nsecret_sauce")
+        expect(passwordText).toContain("Password for all users:")
+        expect(passwordText).toContain(CREDENTIALS.password)
     });
 })
 
-xdescribe('Login Test', () => {
+describe('Login Test', () => {
```
Comment (axboory) [inline test/specs/login.js]: This describe would be better renamed 'Login Flow tests' or 'Login Positive Tests'

**Review by axboory — CHANGES_REQUESTED** [review]

File: test/helper/dataModel.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+exports.ProductPrices = {
+    backpack: '$29.99',
+    bikeLight: '$9.99',
+    boltShirt: '$15.99',
+    fleeceJacket: '$49.99',
+    onesie: '$7.99',
+    redShirt: '$15.99'
+}
```
Comment (axboory) [inline test/helper/dataModel.js]: add a new line at the end of file

File: test/pageobjects/ProductsPage.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
+const {PageObject} = require("./PageObject");
+
+class ProductsPage extends PageObject {
+
```
Comment (axboory) [inline test/pageobjects/ProductsPage.js:4]: add a url getter and the open method.

File: test/pageobjects/LoginPage.js [hunk 6]
```diff
@@ -17,59 +16,51 @@ class LoginPage extends PageObject {
...
+        return browser.$('div.login_password')
 
     }
 
     // get titlePage() {
     //     // browser.url(this.url)
     //     // const $title = browser.getTitle()
-    //     return $('head > title')
+    //     return browser.$('head > title')
     // }
```
Comment (axboory) [inline test/pageobjects/LoginPage.js]: remove if not necessary

File: test/helper/inventoryData.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+exports.ProductPrices = {
+    backpack: '$29.99',
+    bikeLight: '$9.99',
+    boltShirt: '$15.99',
+    fleeceJacket: '$49.99',
+    onesie: '$7.99',
+    redShirt: '$15.99'
+}
```
Comment (axboory) [inline test/helper/inventoryData.js]: add new line to the end of file

File: test/pageobjects/ProductsPage.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        let productsNamesText = []
+        const productNames = await this.allItemsName;
+        for (let i = 0; i < productNames.length; i++){
+            productsNamesText.push(await productNames[i].getText())
+        }
+        return productsNamesText
+    }
+}
+
+module.exports = new ProductsPage();
```
Comment (axboory) [inline test/pageobjects/ProductsPage.js]: add new line to the end of file

File: test/specs/products.js [hunk 9]
```diff
@@ -0,0 +1,112 @@
+const LoginPage = require("../pageobjects/LoginPage");
+const InventoryPage = require("../pageobjects/ProductsPage");
+const {CREDENTIALS} = require("../helper/credentials");
+const {ImageSource, ProductPrices} = require("../helper/inventoryData");
+const {ProductNames} = require("../helper/inventoryData");
+
+describe('Login Test', () => {
```
Comment (axboory) [inline test/specs/products.js]: Login Test?

File: test/specs/products.js [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+
+        let dictValues = Object.values(ProductPrices)
+        // expectedProductPrices.push(price)
+        console.log('expected prices: ', dictValues)
+    });
+
+
+
+
+
```
Comment (axboory) [inline test/specs/products.js]: remove useless space

File: test/specs/products.js [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+        let sortedProductsPrices = []
+        for(let i = 0; i < productsPrices.length; i++){
+            sortedProductsPrices.push(Number(productsPrices[i].replace('$','')))
+        }
+        sortedProductsPrices.sort((a,b)=>a-b)
+        console.log('sorted: ', sortedProductsPrices)
+
+        let dictValues = Object.values(ProductPrices)
+        // expectedProductPrices.push(price)
```
Comment (axboory) [inline test/specs/products.js]: remove unnecessary comment lines and console.log

File: test/specs/login.js [hunk 12]
```diff
@@ -1,128 +1,134 @@
 const LoginPage = require('../pageobjects/LoginPage');
+const InventoryPage = require("../pageobjects/ProductsPage");
 const {CREDENTIALS} = require("../helper/credentials");
+const {ImageSource} = require("../helper/inventoryData");
 
-xdescribe('Browser Tests', () => {
+describe('Login Page UI Tests', () => {
 
     it('should have the "Swag Labs" title', async () => {
         await LoginPage.open();
```
Comment (axboory) [inline test/specs/login.js]: try to use a beforeAll/beforeEach where you load the login page before running the tests so you don't have to write this line every time

File: test/specs/login.js [hunk 13]
```diff
@@ -1,128 +1,134 @@
...
+    });
+
+    it('should not login with wrong username and wrong password', async () => {
+        await LoginPage.open();
+        await LoginPage.login(`test`, 'test');
+
+        expect(await LoginPage.lockedMessage.getText()).toEqual('Epic sadface: Username and password do not match any user in this service')
+
     });
 })
```
Comment (axboory) [inline test/specs/login.js]: add a new line to the end of file

**Review by axboory — CHANGES_REQUESTED** [review]

File: test/specs/products.js (hunk 11 above)
Comment (LiviuAC) [reply test/specs/products.js]: the TC is still under develop. please ignore it for now.

File: test/pageobjects/PageObject.js [hunk 14]
```diff
@@ -8,19 +8,68 @@ class PageObject {
...
+    //TODO: sa adaug side menu elems si logout function
 }
 
-exports.PageObject = PageObject
+
+exports.PageObject = PageObject;
 // exports pt clase, export pt variabile (const, let)
-// npx wdio run ./wdio.conf.js pt a rula testele
\ No newline at end of file
+// npx wdio run ./wdio.conf.js pt a rula testele
```
Comment (axboory) [inline test/pageobjects/PageObject.js]: move the running instructions in ReadMe file

File: test/helper/inventoryData.js [hunk 15]
```diff
@@ -0,0 +1,31 @@
+exports.URLS = {
+    login: 'https://www.saucedemo.com/',
+    products: 'https://www.saucedemo.com/inventory.html'
+}
```
Comment (axboory) [inline test/helper/inventoryData.js]: it's not necessary to save the urls like this. When you want to check if the page has the correct url just verify the 
expect(await browser.getUrl()).toContain(ProductsPage.url);

File: test/specs/products.js [hunk 16]
```diff
@@ -0,0 +1,163 @@
...
+        expect(productsNamesText).toContain(ProductNames.redShirt);
+    });
+
+    it("should display all the 6 products images", async () => {
+        let productsImagesSource = [];
+
+        const productImage = await ProductsPage.allItemsImages;
+        for (let i = 0; i < productImage.length; i++) {
+            productsImagesSource.push(await productImage[i].getAttribute("src"));
+        }
```
Comment (axboory) [inline test/specs/products.js]: maybe try to create a method for extracting images, names and prices.

File: test/helper/beforeAfterEach.js [hunk 17]
```diff
@@ -0,0 +1,19 @@
...
+
+    beforeEach(async function () {
+        await LoginPage.open();
+        await LoginPage.login(CREDENTIALS.standard, CREDENTIALS.password);
+    });
+
+    afterEach(async function () {
+        await ProductsPage.logout();
+    });
+}
```
Comment (axboory) [inline test/helper/beforeAfterEach.js]: what is this file for?

**Review by axboory — CHANGES_REQUESTED** [review]

File: test/pageobjects/PageObject.js (hunk 14 above)
Comment (LiviuAC) [reply test/pageobjects/PageObject.js]: the running instructions have been added in the ReadMe file

File: test/helper/inventoryData.js (hunk 15 above)
Comment (LiviuAC) [reply test/helper/inventoryData.js]: done

File: test/specs/products.js (hunk 16 above)
Comment (LiviuAC) [reply test/specs/products.js]: the method "productData()" has been created in the ProductPage file

File: test/helper/beforeAfterEach.js (hunk 17 above)
Comment (LiviuAC) [reply test/helper/beforeAfterEach.js]: it has been used in order to test something, but it has been deleted

File: test/pageobjects/ProductsPage.js [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+            case 'images':
+                let productsImagesSource = [];
+                const productImage = await this.allItemsImages;
+                for (let i = 0; i < productImage.length; i++) {
+                    productsImagesSource.push(await productImage[i].getAttribute("src"));
+                }
+                return productsImagesSource
+            case 'details':
+                let productsDetailsText = [];
+                const productDetail = await this.allItemsDescriptions;
```
Comment (axboory) [inline test/pageobjects/ProductsPage.js:87]: `productDetail` is already the parameter. change the name of either one.

**Review by axboory — APPROVED** [review]

File: test/pageobjects/ProductsPage.js (hunk 18 above)
Comment (LiviuAC) [reply test/pageobjects/ProductsPage.js:87]: the parameter has been changed into 'data'

**Pull request merged**